it/repositories/MongoJourneyAnswersRepositoryISpec.scala

```diff
@@ -16,123 +16,97 @@
 
 package repositories
 
-import config.AppConfig
+import models.common.JourneyName
+import models.common.JourneyStatus._
 import models.database.JourneyAnswers
-import org.mongodb.scala.model.Filters
-import org.scalatest.BeforeAndAfterEach
 import org.scalatest.matchers.should.Matchers
 import org.scalatest.wordspec.AnyWordSpec
+import org.scalatest.{BeforeAndAfterEach, OptionValues}
 import org.scalatestplus.play.guice.GuiceOneAppPerSuite
-import play.api.Application
-import play.api.inject.bind
-import play.api.inject.guice.GuiceApplicationBuilder
 import play.api.libs.json.Json
 import play.api.test.Helpers.{await, defaultAwaitTimeout}
-import play.api.test.Injecting
 import support.MongoTestSupport
 import uk.gov.hmrc.mongo.test.MongoSupport
+import utils.BaseSpec._
 
-import java.time._
-import java.util.concurrent.TimeUnit
+import java.time.Duration
 import scala.concurrent.ExecutionContext.Implicits.global
-import scala.concurrent.Future
+import scala.concurrent.duration._
 
 class MongoJourneyAnswersRepositoryISpec
     extends AnyWordSpec
     with Matchers
     with MongoSupport
     with MongoTestSupport[JourneyAnswers]
-    with Injecting
     with BeforeAndAfterEach
-    with GuiceOneAppPerSuite {
+    with GuiceOneAppPerSuite
+    with OptionValues {
 
-  private val now   = Instant.parse("2022-01-01T22:02:03.000Z")
-  private val clock = Clock.fixed(now, ZoneOffset.UTC)
+  private val now   = mkNow()
+  private val clock = mkClock(now)
 
-  override lazy val app: Application = new GuiceApplicationBuilder()
-    .overrides(bind[Clock].to(clock))
-    .configure("mongodb.timeToLive" -> "2days")
-    .build()
-
-  private val appConfig   = inject[AppConfig]
-  override val repository = new MongoJourneyAnswersRepository(mongoComponent, appConfig, clock)
-
-  private val someOldTimestamp   = Instant.parse("2022-01-01T21:02:03.000Z")
-  private val id                 = "some_id"
-  private val someJourneyAnswers = JourneyAnswers(id, Json.obj("field" -> "value"), someOldTimestamp)
+  override val repository = new MongoJourneyAnswersRepository(mongoComponent, clock)
 
   override def beforeEach(): Unit = {
-    super.beforeEach()
-    await(removeAll())
+    clock.reset(now)
+    await(removeAll(repository.collection))
   }
 
-  "MongoJourneyAnswersRepository" when {
-    "initialised" must {
-      "include an TTL index for the `lastUpdated` field (where expiry is set through the app config)" in {
-        checkIndex(indexWithField("lastUpdated")) { indexModel =>
-          indexModel.getOptions.getExpireAfter(TimeUnit.DAYS) shouldBe 2
-        }
-      }
+  "upsertData" should {
+    "insert a new journey answers in in-progress status and calculate dates" in {
```

Comment:
it will insert a new row and set the status to in-progress

and calculate dates (we have lots of dates, createdAt, expireAt, updatedAt)

I didn't want to be too specific to not build tets hard to refactor later (if we remove /add a new field with date) we don't need to change the name of the test